dtos/schedulejob.go

```diff
@@ -0,0 +1,296 @@
+}
+
+// Validate satisfies the Validator interface
+func (s *ScheduleJob) Validate() error {
+	err := common.Validate(s)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "invalid ScheduleJob.", err)
+	}
+
+	err = s.Definition.Validate()
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "invalid ScheduleDef.", err)
+	}
+
+	for _, action := range s.Actions {
+		err = action.Validate()
+		if err != nil {
+			return errors.NewCommonEdgeX(errors.KindContractInvalid, "invalid ScheduleAction.", err)
+		}
+	}
+
+	return nil
+}
+
+type ScheduleDef struct {
+	Type string `json:"type" validate:"oneof='Duration' 'Cron'"`
+
+	DurationScheduleDef `json:",inline" validate:"-"`
+	CronScheduleDef     `json:",inline" validate:"-"`
+}
+
+// Validate satisfies the Validator interface
+func (s *ScheduleDef) Validate() error {
+	err := common.Validate(s)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "invalid ScheduleDef.", err)
+	}
+
+	switch s.Type {
+	case common.DefDuration:
+		err = common.Validate(s.DurationScheduleDef)
+		if err != nil {
+			return errors.NewCommonEdgeX(errors.KindContractInvalid, "invalid DurationScheduleDef.", err)
+		}
+	case common.DefCron:
+		err = common.Validate(s.CronScheduleDef)
+		if err != nil {
+			return errors.NewCommonEdgeX(errors.KindContractInvalid, "invalid CronScheduleDef.", err)
+		}
+	}
+
+	return nil
+}
+
+type DurationScheduleDef struct {
+	Duration int64 `json:"duration" validate:"required"`
+}
+
+type CronScheduleDef struct {
+	Crontab string `json:"crontab" validate:"required"`
+}
+
+type ScheduleAction struct {
+	Type        string `json:"type" validate:"oneof='MessageBus' 'REST' 'DeviceControl'"`
+	ContentType string `json:"contentType,omitempty"`
+	Payload     []byte `json:"payload,omitempty"`
+
+	MessageBusAction    `json:",inline" validate:"-"`
+	RESTAction          `json:",inline" validate:"-"`
+	DeviceControlAction `json:",inline" validate:"-"`
+}
+
+func (s *ScheduleAction) Validate() error {
+	err := common.Validate(s)
+	if err != nil {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, "invalid ScheduleAction.", err)
+	}
+
+	switch s.Type {
+	case common.ActionMessageBus:
+		err = common.Validate(s.MessageBusAction)
+		if err != nil {
+			return errors.NewCommonEdgeX(errors.KindContractInvalid, "invalid MessageBusAction.", err)
+		}
+	case common.ActionREST:
+		err = common.Validate(s.RESTAction)
+		if err != nil {
+			return errors.NewCommonEdgeX(errors.KindContractInvalid, "invalid RESTAction.", err)
+		}
+	case common.ActionDeviceControl:
+		err = common.Validate(s.DeviceControlAction)
+		if err != nil {
+			return errors.NewCommonEdgeX(errors.KindContractInvalid, "invalid DeviceControlAction.", err)
+		}
+	}
+
+	return nil
+}
+
+type MessageBusAction struct {
+	Topic string `json:"topic" validate:"required"`
+}
+
+type RESTAction struct {
+	Address string `json:"address" validate:"required"`
```

Comment:
add InjectEdgeXAuth to RESTAction
https://github.com/edgexfoundry/go-mod-core-contracts/blob/efd56194609a555529ad18af082fdd308362e621/dtos/address.go#L55

If it's true, it means the endpoint is EdgeX services, and add inject EdgeX JWT to the REST call